### DISCUSSION TITLE
gmx editconf failed when querying com_traj.xtc

### DISCUSSION THREAD

**Discussion by afwer1231** [Valdes-Tresanco-MS/gmx_MMPBSA#391] [General]
dear Valdes:
I am trying to calculate the mmpbsa between a cyclodextrin and a ligand with LPH atoms. After following this tutorial(https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA/tree/master/examples/Protein_ligand_LPH_atoms_CHARMMff) to prepare my input files I use the following command:
mpirun -np 2 gmx_MMPBSA -O -i mmpbsa.in -cs str_noLP.pdb -ci index.ndx -cg 7 3 -ct com_traj.xtc -cp topol.top -o FINAL_RESULTS_MMPBSA.dat -eo FINAL_RESULTS_MMPBSA.csv
However, it gives me this error and the job stops :
[INFO   ] Checking mmpbsa.in input file...
[INFO   ] Checking mmpbsa.in input file...Done.

[INFO   ] Checking external programs...
[INFO   ] cpptraj found! Using /home/ai/miniconda3/envs/gmxMMPBSA/bin/cpptraj
[INFO   ] tleap found! Using /home/ai/miniconda3/envs/gmxMMPBSA/bin/tleap
[INFO   ] parmchk2 found! Using /home/ai/miniconda3/envs/gmxMMPBSA/bin/parmchk2
[INFO   ] sander found! Using /home/ai/miniconda3/envs/gmxMMPBSA/bin/sander
[INFO   ] Using GROMACS version > 5.x.x!
[INFO   ] gmx found! Using /usr/bin/gmx
[INFO   ] Checking external programs...Done.

[INFO   ] Building AMBER topologies from GROMACS files...
[INFO   ] Get PDB files from GROMACS structures files...
[INFO   ] Making gmx_MMPBSA index for complex...
[INFO   ] Normal Complex: Saving group ligand_receptor (7_3) in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_COM.pdb
[ERROR  ] MMPBSA_Error

/usr/bin/gmx editconf failed when querying com_traj.xtc

Check the gmx_MMPBSA.log file to report the problem.
tets
  File "/home/ai/miniconda3/envs/gmxMMPBSA/bin/gmx_MMPBSA", line 8, in <module>
    sys.exit(gmxmmpbsa())
  File "/home/ai/miniconda3/envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/app.py", line 98, in gmxmmpbsa
    app.make_prmtops()
  File "/home/ai/miniconda3/envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/main.py", line 687, in make_prmtops
    self.FILES.mutant_receptor_prmtop, self.FILES.mutant_ligand_prmtop) = maketop.buildTopology()
  File "/home/ai/miniconda3/envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/make_top.py", line 123, in buildTopology
    self.gmx2pdb()
  File "/home/ai/miniconda3/envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/make_top.py", line 289, in gmx2pdb
    GMXMMPBSA_ERROR('%s failed when querying %s' % (' '.join(comprog), self.FILES.complex_trajs[0]))
  File "/home/ai/miniconda3/envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/exceptions.py", line 171, in __init__
    raise exc('\n\n' + msg + '\n\nCheck the gmx_MMPBSA.log file to report the problem.')
MMPBSA_Error:

/usr/bin/gmx editconf failed when querying com_traj.xtc

Check the gmx_MMPBSA.log file to report the problem.
Error occurred on rank 0.
Exiting. All files have been retained.
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0

I will be glad to see you guide me on how to get this resolved .

Regards,
fengwei ai

**gmx_MMPBSA.log**

[INFO   ] Starting gmx_MMPBSA v1.6.1
[DEBUG  ] WDIR          : /home/ai/rmcd12-3dfnb(charm36)-gmxMMPDSA分析
[DEBUG  ] AMBERHOME     : /home/ai/miniconda3/envs/gmxMMPBSA
[DEBUG  ] PYTHON EXE    : /home/ai/miniconda3/envs/gmxMMPBSA/bin/python
[DEBUG  ] PYTHON VERSION: 3.9.12 | packaged by conda-forge | (main, Mar 24 2022, 23:25:59) [GCC 10.3.0]
[DEBUG  ] MPI           : /home/ai/miniconda3/envs/gmxMMPBSA/bin/mpirun
[DEBUG  ] ParmEd        : 3.4.3
[DEBUG  ] OS PLATFORM   : Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.31
[DEBUG  ] OS SYSTEM     : Linux
[DEBUG  ] OS VERSION    : #1 SMP Fri Jan 27 02:56:13 UTC 2023
[DEBUG  ] OS PROCESSOR  : x86_64

[INFO   ] Command-line
  mpirun -np 2 gmx_MMPBSA -O -i mmpbsa.in -cs str_noLP.pdb -ci index.ndx -cg 7 3 -ct com_traj.xtc -cp topol.top -o FINAL_RESULTS_MMPBSA.dat -eo FINAL_RESULTS_MMPBSA.csv

[DEBUG  ] |Input file:
[DEBUG  ] |--------------------------------------------------------------
[DEBUG  ] |Sample input file for GB calculation
[DEBUG  ] |#This input file is meant to show only that gmx_MMPBSA works. Although, we tried to use the input files as recommended in the
[DEBUG  ] |#Amber manual, some parameters have been changed to perform more expensive calculations in a reasonable amount of time. Feel free to change the parameters
[DEBUG  ] |#according to what is better for your system.
[DEBUG  ] |
[DEBUG  ] |&general
[DEBUG  ] |sys_name="Prot-Lig-ST",
[DEBUG  ] |startframe=5,
[DEBUG  ] |endframe=9,
[DEBUG  ] |solvated_trajectory=0,
[DEBUG  ] |# In gmx_MMPBSA v1.5.0 we have added a new PB radii set named charmm_radii. This radii set should be used only
[DEBUG  ] |# with systems prepared with CHARMM force fields. Uncomment the line below to use charmm_radii set
[DEBUG  ] |#PBRadii=7,
[DEBUG  ] |/
[DEBUG  ] |&pb
[DEBUG  ] |# radiopt=0 is recommended which means using radii from the prmtop file
[DEBUG  ] |# for both the PB calculation and for the NP calculation
[DEBUG  ] |
[DEBUG  ] |istrng=0.15, fillratio=1.250, radiopt=0, inp=1,
[DEBUG  ] |/--------------------------------------------------------------
[DEBUG  ] 
[INFO   ] Checking mmpbsa.in input file...
[INFO   ] Checking mmpbsa.in input file...Done.

[INFO   ] Checking external programs...
[INFO   ] cpptraj found! Using /home/ai/miniconda3/envs/gmxMMPBSA/bin/cpptraj
[INFO   ] tleap found! Using /home/ai/miniconda3/envs/gmxMMPBSA/bin/tleap
[INFO   ] parmchk2 found! Using /home/ai/miniconda3/envs/gmxMMPBSA/bin/parmchk2
[INFO   ] sander found! Using /home/ai/miniconda3/envs/gmxMMPBSA/bin/sander
[INFO   ] Using GROMACS version > 5.x.x!
[INFO   ] gmx found! Using /usr/bin/gmx
[INFO   ] Checking external programs...Done.

[INFO   ] Building AMBER topologies from GROMACS files...
[INFO   ] Get PDB files from GROMACS structures files...
[INFO   ] Making gmx_MMPBSA index for complex...
[DEBUG  ] Running command: echo -e "name 7 GMXMMPBSA_REC\n name 3 GMXMMPBSA_LIG\n  7 | 3\n q\n" | /usr/bin/gmx make_ndx -n index.ndx -o _GMXMMPBSA_COM_index.ndx -f str_noLP.pdb
[DEBUG  ]              :-) GROMACS - gmx make_ndx, 2020.1-Ubuntu-2020.1-1 (-:
[DEBUG  ] 
[DEBUG  ]                             GROMACS is written by:
[DEBUG  ]      Emile Apol      Rossen Apostolov      Paul Bauer     Herman J.C. Berendsen
[DEBUG  ]     Par Bjelkmar      Christian Blau   Viacheslav Bolnykh     Kevin Boyd    
[DEBUG  ]  Aldert van Buuren   Rudi van Drunen     Anton Feenstra       Alan Gray     
[DEBUG  ]   Gerrit Groenhof     Anca Hamuraru    Vincent Hindriksen  M. Eric Irrgang  
[DEBUG  ]   Aleksei Iupinov   Christoph Junghans     Joe Jordan     Dimitrios Karkoulis
[DEBUG  ]     Peter Kasson        Jiri Kraus      Carsten Kutzner      Per Larsson    
[DEBUG  ]   Justin A. Lemkul    Viveca Lindahl    Magnus Lundborg     Erik Marklund   
[DEBUG  ]     Pascal Merz     Pieter Meulenhoff    Teemu Murtola       Szilard Pall   
[DEBUG  ]     Sander Pronk      Roland Schulz      Michael Shirts    Alexey Shvetsov  
[DEBUG  ]    Alfons Sijbers     Peter Tieleman      Jon Vincent      Teemu Virolainen 
[DEBUG  ]  Christian Wennberg    Maarten Wolf      Artem Zhmurov   
[DEBUG  ]                            and the project leaders:
[DEBUG  ]         Mark Abraham, Berk Hess, Erik Lindahl, and David van der Spoel
[DEBUG  ] 
[DEBUG  ] Copyright (c) 1991-2000, University of Groningen, The Netherlands.
[DEBUG  ] Copyright (c) 2001-2019, The GROMACS development team at
[DEBUG  ] Uppsala University, Stockholm University and
[DEBUG  ] the Royal Institute of Technology, Sweden.
[DEBUG  ] check out http://www.gromacs.org for more information.
[DEBUG  ] 
[DEBUG  ] GROMACS is free software; you can redistribute it and/or modify it
[DEBUG  ] under the terms of the GNU Lesser General Public License
[DEBUG  ] as published by the Free Software Foundation; either version 2.1
[DEBUG  ] of the License, or (at your option) any later version.
[DEBUG  ] 
[DEBUG  ] GROMACS:      gmx make_ndx, version 2020.1-Ubuntu-2020.1-1
[DEBUG  ] Executable:   /usr/bin/gmx
[DEBUG  ] Data prefix:  /usr
[DEBUG  ] Working dir:  /home/ai/rmcd12-3dfnb(charm36)-gmxMMPDSA分析
[DEBUG  ] Command line:
[DEBUG  ]   gmx make_ndx -n index.ndx -o _GMXMMPBSA_COM_index.ndx -f str_noLP.pdb
[DEBUG  ] 
[DEBUG  ] 
[DEBUG  ] Reading structure file
[DEBUG  ] 
[DEBUG  ] GROMACS reminds you: "Marie, you're looking more radiant every day!" (Pierre Curie)
[DEBUG  ] 
[DEBUG  ] Going to read 1 old index file(s)
[DEBUG  ] 
[DEBUG  ]   0 System              :  7193 atoms
[DEBUG  ]   1 Other               :   230 atoms
[DEBUG  ]   2 fnb                 :    47 atoms
[DEBUG  ]   3 receptor            :   183 atoms
[DEBUG  ]   4 Water               :  6963 atoms
[DEBUG  ]   5 SOL                 :  6963 atoms
[DEBUG  ]   6 non-Water           :   230 atoms
[DEBUG  ]   7 ligand              :    46 atoms
[DEBUG  ]   8 receptor_ligand     :   229 atoms
[DEBUG  ] 
[DEBUG  ]  nr : group      '!': not  'name' nr name   'splitch' nr    Enter: list groups
[DEBUG  ]  'a': atom       '&': and  'del' nr         'splitres' nr   'l': list residues
[DEBUG  ]  't': atom type  '|': or   'keep' nr        'splitat' nr    'h': help
[DEBUG  ]  'r': residue              'res' nr         'chain' char
[DEBUG  ]  "name": group             'case': case sensitive           'q': save and quit
[DEBUG  ]  'ri': residue index
[DEBUG  ] 
[DEBUG  ] > 
[DEBUG  ] 
[DEBUG  ] > 
[DEBUG  ] 
[DEBUG  ] > 
[DEBUG  ] Copied index group 7 'GMXMMPBSA_REC'
[DEBUG  ] Copied index group 3 'GMXMMPBSA_LIG'
[DEBUG  ] Merged two groups with OR: 46 183 -> 229
[DEBUG  ] 
[DEBUG  ]   9 GMXMMPBSA_REC_GMXMMPBSA_LIG:   229 atoms
[DEBUG  ] 
[DEBUG  ] > 
[INFO   ] Normal Complex: Saving group ligand_receptor (7_3) in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_COM.pdb
[DEBUG  ] Running command: echo -e "GMXMMPBSA_REC_GMXMMPBSA_LIG"| /usr/bin/gmx editconf -f str_noLP.pdb -n _GMXMMPBSA_COM_index.ndx -o _GMXMMPBSA_COM.pdb
[DEBUG  ]              :-) GROMACS - gmx editconf, 2020.1-Ubuntu-2020.1-1 (-:
[DEBUG  ] 
[DEBUG  ]                             GROMACS is written by:
[DEBUG  ]      Emile Apol      Rossen Apostolov      Paul Bauer     Herman J.C. Berendsen
[DEBUG  ]     Par Bjelkmar      Christian Blau   Viacheslav Bolnykh     Kevin Boyd    
[DEBUG  ]  Aldert van Buuren   Rudi van Drunen     Anton Feenstra       Alan Gray     
[DEBUG  ]   Gerrit Groenhof     Anca Hamuraru    Vincent Hindriksen  M. Eric Irrgang  
[DEBUG  ]   Aleksei Iupinov   Christoph Junghans     Joe Jordan     Dimitrios Karkoulis
[DEBUG  ]     Peter Kasson        Jiri Kraus      Carsten Kutzner      Per Larsson    
[DEBUG  ]   Justin A. Lemkul    Viveca Lindahl    Magnus Lundborg     Erik Marklund   
[DEBUG  ]     Pascal Merz     Pieter Meulenhoff    Teemu Murtola       Szilard Pall   
[DEBUG  ]     Sander Pronk      Roland Schulz      Michael Shirts    Alexey Shvetsov  
[DEBUG  ]    Alfons Sijbers     Peter Tieleman      Jon Vincent      Teemu Virolainen 
[DEBUG  ]  Christian Wennberg    Maarten Wolf      Artem Zhmurov   
[DEBUG  ]                            and the project leaders:
[DEBUG  ]         Mark Abraham, Berk Hess, Erik Lindahl, and David van der Spoel
[DEBUG  ] 
[DEBUG  ] Copyright (c) 1991-2000, University of Groningen, The Netherlands.
[DEBUG  ] Copyright (c) 2001-2019, The GROMACS development team at
[DEBUG  ] Uppsala University, Stockholm University and
[DEBUG  ] the Royal Institute of Technology, Sweden.
[DEBUG  ] check out http://www.gromacs.org for more information.
[DEBUG  ] 
[DEBUG  ] GROMACS is free software; you can redistribute it and/or modify it
[DEBUG  ] under the terms of the GNU Lesser General Public License
[DEBUG  ] as published by the Free Software Foundation; either version 2.1
[DEBUG  ] of the License, or (at your option) any later version.
[DEBUG  ] 
[DEBUG  ] GROMACS:      gmx editconf, version 2020.1-Ubuntu-2020.1-1
[DEBUG  ] Executable:   /usr/bin/gmx
[DEBUG  ] Data prefix:  /usr
[DEBUG  ] Working dir:  /home/ai/rmcd12-3dfnb(charm36)-gmxMMPDSA分析
[DEBUG  ] Command line:
[DEBUG  ]   gmx editconf -f str_noLP.pdb -n _GMXMMPBSA_COM_index.ndx -o _GMXMMPBSA_COM.pdb
[DEBUG  ] 
[DEBUG  ] 
[DEBUG  ] Select a group for output:
[DEBUG  ] Group     0 (         System) has  7193 elements
[DEBUG  ] Group     1 (          Other) has   230 elements
[DEBUG  ] Group     2 (            fnb) has    47 elements
[DEBUG  ] Group     3 (  GMXMMPBSA_LIG) has   183 elements
[DEBUG  ] Group     4 (          Water) has  6963 elements
[DEBUG  ] Group     5 (            SOL) has  6963 elements
[DEBUG  ] Group     6 (      non-Water) has   230 elements
[DEBUG  ] Group     7 (  GMXMMPBSA_REC) has    46 elements
[DEBUG  ] Group     8 (receptor_ligand) has   229 elements
[DEBUG  ] Group     9 (GMXMMPBSA_REC_GMXMMPBSA_LIG) has   229 elements
[DEBUG  ] Select a group: 
[ERROR  ] MMPBSA_Error 

/usr/bin/gmx editconf failed when querying com_traj.xtc

Check the gmx_MMPBSA.log file to report the problem.

**Comment by marioernestovaldes** [comment]
This issue typically occurs when the index file is not the correct one... there are plenty of examples in in google group on how to solve this issue

**Comment by afwer1231** [comment]
Thanks for your reply. Yes, the problem has been solved. I would like to ask how to set the output unit as kJ/mol.




With best wishes,


Fengwei Ai
School of Pharmacy, Xuzhou Medical University
E-Mail: ***@***.***
---- Replied Message ----
| From | Mario E. ***@***.***> |
| Date | 11/30/2023 08:05 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [Valdes-Tresanco-MS/gmx_MMPBSA] gmx editconf failed when querying com_traj.xtc (Discussion #391) |

This issue typically occurs when the index file is not the correct one... there are plenty of examples in in google group on how to solve this issue

**Comment by marioernestovaldes** [comment]
At the moment the option to transform to Kj/mol is not available...

**Comment by afwer1231** [comment]
OK.




With best wishes,


Fengwei Ai
School of Pharmacy, Xuzhou Medical University
E-Mail: ***@***.***
---- Replied Message ----
| From | Mario E. ***@***.***> |
| Date | 2/23/2024 11:33 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [Valdes-Tresanco-MS/gmx_MMPBSA] gmx editconf failed when querying com_traj.xtc (Discussion #391) |

At the moment the option to transform to Kj/mol is not available...

**Comment by Zaiabari** [comment]
Hi There,
I have the same issue: 
MMPBSA_Error: 

/usr/local/gromacs/bin/gmx editconf failed when querying com_traj.xtc

how did you manage to solve it?
Thanks

**Comment by afwer1231** [comment]
Yes, my error problem has been solved. In my case, the following command was executed incorrectly and an error similar to yours occurred

mpirun -np 2 gmx_MMPBSA -O -i mmpbsa.in -cs str_noLP.pdb -ci index.ndx -cg 2 3 -ct com_traj.xtc -cp topol.top -o FINAL_RESULTS_MMPBSA.dat -eo FINAL_RESULTS_MMPBSA.csv  

You need to generate a new index with the str_noLP.pdb file,as follows:

gmx make_ndx -f str_noLP.pdb -o m_index.ndx

mpirun -np 16 gmx_MMPBSA -O -i mmpbsa.in -cs str_noLP.pdb -ci m_index.ndx -cg 2 3 -ct com_traj.xtc -cp topol.top -o FINAL_RESULTS_MMPBSA.dat -eo FINAL_RESULTS_MMPBSA.csv


Problem solved.





With best wishes,


Fengwei Ai
School of Pharmacy, Xuzhou Medical University
E-Mail: ***@***.***, ***@***.***
---- Replied Message ----
| From | Ali Abdolahzadeh ***@***.***> |
| Date | 1/11/2025 01:04 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [Valdes-Tresanco-MS/gmx_MMPBSA] gmx editconf failed when querying com_traj.xtc (Discussion #391) |

Hi There,
I have the same issue:
MMPBSA_Error:

/usr/local/gromacs/bin/gmx editconf failed when querying com_traj.xtc

how did you manage to solve it?
Thanks
